pkg/prometheus/server/operator.go

```diff
@@ -210,91 +179,18 @@ func New(ctx context.Context, conf operator.Config, logger log.Logger, r prometh
 		r,
 	)
 
-	c.promInfs, err = informers.NewInformersForResource(
-		informers.NewMonitoringInformerFactories(
-			c.config.Namespaces.PrometheusAllowList,
-			c.config.Namespaces.DenyList,
-			mclient,
-			resyncPeriod,
-			func(options *metav1.ListOptions) {
-				options.LabelSelector = c.config.PromSelector
-			},
-		),
-		monitoringv1.SchemeGroupVersion.WithResource(monitoringv1.PrometheusName),
-	)
-	if err != nil {
-		return nil, errors.Wrap(err, "error creating prometheus informers")
-	}
-
 	var promStores []cache.Store
 	for _, informer := range c.promInfs.GetInformers() {
 		promStores = append(promStores, informer.Informer().GetStore())
 	}
 	c.metrics.MustRegister(prompkg.NewCollectorForStores(promStores...))
 
-	c.smonInfs, err = informers.NewInformersForResource(
-		informers.NewMonitoringInformerFactories(
-			c.config.Namespaces.AllowList,
-			c.config.Namespaces.DenyList,
-			mclient,
-			resyncPeriod,
-			nil,
-		),
-		monitoringv1.SchemeGroupVersion.WithResource(monitoringv1.ServiceMonitorName),
-	)
-	if err != nil {
-		return nil, errors.Wrap(err, "error creating servicemonitor informers")
-	}
-
-	c.pmonInfs, err = informers.NewInformersForResource(
-		informers.NewMonitoringInformerFactories(
-			c.config.Namespaces.AllowList,
-			c.config.Namespaces.DenyList,
-			mclient,
-			resyncPeriod,
-			nil,
-		),
-		monitoringv1.SchemeGroupVersion.WithResource(monitoringv1.PodMonitorName),
-	)
-	if err != nil {
-		return nil, errors.Wrap(err, "error creating podmonitor informers")
-	}
-
-	c.probeInfs, err = informers.NewInformersForResource(
-		informers.NewMonitoringInformerFactories(
-			c.config.Namespaces.AllowList,
-			c.config.Namespaces.DenyList,
-			mclient,
-			resyncPeriod,
-			nil,
-		),
-		monitoringv1.SchemeGroupVersion.WithResource(monitoringv1.ProbeName),
-	)
-	if err != nil {
-		return nil, errors.Wrap(err, "error creating probe informers")
-	}
-
-	if c.scrapeConfigSupported {
```

Comment:
Here we have a different behaviour. In the Agent operator we keep it but in the server one we remove it. Is that on purpose? see https://github.com/prometheus-operator/prometheus-operator/pull/5499/files#r1224188515

Reply:
don't know what do you mean.

Reply:
Reviewing this comment, I actually added a comment on removed code, please disregard, thanks!